### PR TITLE
feat: improve contact endpoints for management front

### PR DESCRIPTION
**Pull request by BettyB979**

### CODE REVIEW

File: src/main/java/fr/insee/survey/datacollectionmanagement/questioning/controller/SurveyUnitCommentController.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        surveyUnitComment.setDate(new Date());
+        SurveyUnitComment newSurveyUnitComment = surveyUnitCommentService.saveSurveyUnitComment(surveyUnitComment);
+        Set<SurveyUnitComment> setSurveyUnitComments = surveyUnit.getSurveyUnitComments();
+        setSurveyUnitComments.add(newSurveyUnitComment);
+        surveyUnit.setSurveyUnitComments(setSurveyUnitComments);
+        surveyUnitService.saveSurveyUnit(surveyUnit);
+        HttpHeaders responseHeaders = new HttpHeaders();
+        responseHeaders.set(HttpHeaders.LOCATION,
+                ServletUriComponentsBuilder.fromCurrentRequest().toUriString());
+        return ResponseEntity.status(HttpStatus.CREATED).headers(responseHeaders)
```
Comment (davdarras) [inline src/main/java/fr/insee/survey/datacollectionmanagement/questioning/controller/SurveyUnitCommentController.java:67]: why setting response headers here ? The method could return void, and set the created http status in an annotation

File: src/main/java/fr/insee/survey/datacollectionmanagement/questioning/controller/SurveyUnitCommentController.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    @Operation(summary = "Create a survey unit comment")
+    @PostMapping(value = Constants.API_SURVEY_UNITS_ID_COMMENT, produces = "application/json", consumes = "application/json")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "201", description = "Created", content = @Content(schema = @Schema(implementation = SurveyUnitCommentInputDto.class))),
+            @ApiResponse(responseCode = "400", description = "Bad request"),
+            @ApiResponse(responseCode = "404", description = "Not found")
+    })
+    public ResponseEntity postSurveyUnitComment(@PathVariable String id, @Valid @RequestBody SurveyUnitCommentInputDto surveyUnitCommentDto) {
+
+        SurveyUnit surveyUnit = surveyUnitService.findbyId(id);
```
Comment (davdarras) [inline src/main/java/fr/insee/survey/datacollectionmanagement/questioning/controller/SurveyUnitCommentController.java:56]: setting entity in the controller, why setting comments in the survey unit object ?

File: src/main/java/fr/insee/survey/datacollectionmanagement/questioning/dto/SearchSurveyUnitDto.java [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package fr.insee.survey.datacollectionmanagement.questioning.dto;
+
+public interface SearchSurveyUnitDto {
+
+    String getIdSu();
+    String getIdentificationCode();
+    String getIdentificationName();
```
Comment (davdarras) [inline src/main/java/fr/insee/survey/datacollectionmanagement/questioning/dto/SearchSurveyUnitDto.java:7]: why an interface for this ?

File: src/main/java/fr/insee/survey/datacollectionmanagement/questioning/controller/SurveyUnitController.java [hunk 4]
```diff
@@ -60,33 +70,44 @@ public Page<SurveyUnitDto> getSurveyUnits(
...
             @ApiResponse(responseCode = "404", description = "Not found"),
             @ApiResponse(responseCode = "400", description = "Bad Request")
     })
-    public Page<SurveyUnitDto> searchSurveyUnits(
-            @RequestParam(required = false) String idSu,
-            @RequestParam(required = false) String identificationCode,
-            @RequestParam(required = false) String identificationName,
+    public Page<SearchSurveyUnitDto> searchSurveyUnits(
+            @RequestParam(required = true) String searchParam,
+            @RequestParam(required = true) @Valid @ValidSurveyUnitParam @Schema(description = "id or code or name")String searchType,
```
Comment (davdarras) [inline src/main/java/fr/insee/survey/datacollectionmanagement/questioning/controller/SurveyUnitController.java:79]: does it work without @Validated on the controller class ?

Comment (davdarras) [inline src/main/java/fr/insee/survey/datacollectionmanagement/questioning/controller/SurveyUnitController.java:79]: It should be working without custom annotations here by giving directly the enum as parameter

Comment (davdarras) [reply src/main/java/fr/insee/survey/datacollectionmanagement/questioning/controller/SurveyUnitController.java:79]: Same goes for the other custom validator annotation

**Review by davdarras — COMMENTED** [review]
Waiting for the 🍺

File: src/main/java/fr/insee/survey/datacollectionmanagement/questioning/dto/SearchSurveyUnitDto.java (hunk 3 above)
Comment (BettyB979) [reply src/main/java/fr/insee/survey/datacollectionmanagement/questioning/dto/SearchSurveyUnitDto.java:7]: Passing through an interface acts as a converter between my SearchSurveyUnitDto and the result of my query.

File: src/main/java/fr/insee/survey/datacollectionmanagement/questioning/controller/SurveyUnitController.java (hunk 4 above)
Comment (BettyB979) [reply src/main/java/fr/insee/survey/datacollectionmanagement/questioning/controller/SurveyUnitController.java:79]: the @validated annotation is present on the controller. And it does not work if I remove it

**Pull request merged**